packages/compass-collection/src/transform-schema-to-field-info.spec.ts

```diff
@@ -0,0 +1,868 @@
+          path: ['tags'],
+          probability: 1.0,
+          types: [
+            {
+              name: 'Array',
+              bsonType: 'Array',
+              path: ['tags'],
+              count: 2,
+              probability: 1.0,
+              values: [['red', 'blue'], ['green']],
+              lengths: [2, 1],
+              averageLength: 1.5,
+              totalCount: 3,
+              types: [
+                {
+                  name: 'String',
+                  bsonType: 'String',
+                  path: ['tags'],
+                  count: 3,
+                  probability: 1.0,
+                  values: ['red', 'blue', 'green'],
+                } as SchemaType,
+              ],
+            } as ArraySchemaType,
+          ],
+        } as SchemaField,
+      ],
+      count: 2,
+    };
+
+    const result = processSchema(schema);
+
+    expect(result).to.deep.equal({
+      'tags[]': {
+        type: 'String',
+        sample_values: ['red', 'blue', 'green'],
+        array_sample_values: [['red', 'blue'], ['green']],
+        probability: 1.0,
+      },
+    });
+  });
+
+  it('handles deeply nested objects (documents)', function () {
+    const schema: Schema = {
+      fields: [
+        {
+          name: 'level1',
+          path: ['level1'],
+          probability: 1.0,
+          types: [
+            {
+              name: 'Document',
+              bsonType: 'Document',
+              path: ['level1'],
+              count: 1,
+              probability: 1.0,
+              fields: [
+                {
+                  name: 'level2',
+                  path: ['level1', 'level2'],
+                  probability: 1.0,
+                  types: [
+                    {
+                      name: 'Document',
+                      bsonType: 'Document',
+                      path: ['level1', 'level2'],
+                      count: 1,
+                      probability: 1.0,
+                      fields: [
+                        {
+                          name: 'value',
+                          path: ['level1', 'level2', 'value'],
+                          probability: 1.0,
+                          types: [
+                            {
+                              name: 'String',
+                              bsonType: 'String',
+                              path: ['level1', 'level2', 'value'],
+                              count: 1,
+                              probability: 1.0,
+                              values: ['deep'],
+                            } as SchemaType,
+                          ],
+                        } as SchemaField,
+                      ],
+                    } as DocumentSchemaType,
+                  ],
+                } as SchemaField,
+              ],
+            } as DocumentSchemaType,
+          ],
+        } as SchemaField,
+      ],
+      count: 1,
+    };
+
+    const result = processSchema(schema);
+
+    expect(result).to.deep.equal({
```

Comment:
ah this is a good example of how the dot notation combined with the constraints we've placed simplifies the parsing